hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java

```diff
@@ -636,61 +654,168 @@ private void insertTokenAccountRevokeKyc(RecordItem recordItem) {
+        for (TokenTransferList transferList : tokenTransfersList) {
+            if (!transferList.getToken().equals(nftId)) {
+                continue;
+            }
+            for (NftTransfer transfer : transferList.getNftTransfersList()) {
+                if (transfer.getSerialNumber() == nftTransfer.getSerialNumber() &&
+                        transfer.getReceiverAccountID().equals(nftTransfer.getReceiverAccountID()) &&
+                        transfer.getSenderAccountID().equals(nftTransfer.getSenderAccountID())) {
+                    return transfer;
+                }
+            }
+        }
+        return null;
+    }
 
-                    entityListener.onNftTransfer(nftTransferDomain);
-                    if (!EntityId.isEmpty(receiverId)) {
-                        transferNftOwnership(consensusTimestamp, serialNumber, tokenId, receiverId);
+    private void insertFungibleTokenTransfers(
+            long consensusTimestamp, TransactionBody body, boolean isTokenDissociate,
+            TokenID tokenId, EntityId entityTokenId, EntityId payerAccountId, List<AccountAmount> tokenTransfers){
+        for (AccountAmount accountAmount : tokenTransfers) {
+            EntityId accountId = EntityId.of(accountAmount.getAccountID());
+            long amount = accountAmount.getAmount();
+            TokenTransfer tokenTransfer = new TokenTransfer();
+            tokenTransfer.setAmount(amount);
+            tokenTransfer.setId(new TokenTransfer.Id(consensusTimestamp, entityTokenId, accountId));
+            tokenTransfer.setIsApproval(false);
+            tokenTransfer.setPayerAccountId(payerAccountId);
+            tokenTransfer.setTokenDissociate(isTokenDissociate);
+
+            // If a record AccountAmount with amount < 0 is not in the body;
+            // but an AccountAmount with the same (TokenID, AccountID) combination is in the body with is_approval=true,
+            // then again set is_approval=true
+            if (accountAmount.getAmount() < 0) {
+
+                // Is the accountAmount from the record also inside a body's transfer list for the given tokenId?
+                AccountAmount accountAmountInsideTransferList =
+                        findAccountAmountInTokenTransferListInsideBody(
+                                accountAmount::equals, tokenId, body);
+                if (accountAmountInsideTransferList == null) {
+
+                    // Is there any account amount inside the body's transfer list for the given tokenId
+                    // with the same accountId as the accountAmount from the record?
+                    AccountAmount accountAmountWithSameAccountIdInsideTransferList =
+                            findAccountAmountInTokenTransferListInsideBody(
+                                    aa -> aa.getAccountID().equals(accountAmount.getAccountID()) && aa.getIsApproval(),
+                                    tokenId, body);
```

Comment:
Perhaps shorten these method and variable names so one line doesn't take up 4. Can use overloaded `findAccountAmount` for both. Can do the same in some other places.

```suggestion
                    AccountAmount accountAmountBody = findAccountAmount(this::isApprovedAccountAmount, tokenId, body);
```

Reply:
Done.